### PR TITLE
Add setup documentation

### PR DESCRIPTION
**Pull request by balthasars**
Adds setup article to pkdown site from gitlab. Already adjusted for newest version.

### CODE REVIEW

File: vignettes/setting_up_dm.Rmd [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  %\VignetteEngine{knitr::rmarkdown}
+editor_options: 
+  chunk_output_type: console
+---
+
+<!-- README.md is generated from README.Rmd. Please edit that file -->
+
+This document describes how you can get your data into a `dm` object. Though this example is also built in already — just run `cdm_nycflights13()` — here, we'll go through the process step by step.
+
+The five tables we are working with contain information on flights from airports of the North-American city of New York to other destinations in the United States in 2013 and are available through the package [`nycflights13`](https://github.com/hadley/nycflights13)). `flights` represents the journeys taken by planes, `airlines` includes transport organizations' names (`name`) and their shortcuts (`carrier`), `airports` details the arrival and departure stations, `weater` the meterological context at each hour, while `planes` describes aircrafts' characteristics. Once we've loaded `nycflights13`, the aforementioned tables are all in our work environment, ready for being accessed.
```
Comment (krlmlr) [inline vignettes/setting_up_dm.Rmd]:
```suggestion
The five tables we are working with contain information on flights from airports of the North-American city of New York to other destinations in the United States in 2013 and are available through the package [`nycflights13`](https://github.com/hadley/nycflights13). `flights` represents the journeys taken by planes, `airlines` includes transport organizations' names (`name`) and their shortcuts (`carrier`), `airports` details the arrival and departure stations, `weather` the meterological context at each hour, while `planes` describes the characteristics of the aircrafts. Once we've loaded `nycflights13`, the aforementioned tables are all in our work environment, ready for access.
```

Maybe use bullet points?

File: vignettes/setting_up_dm.Rmd [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+# Adding Tables
+
+First, we will tell `dm` what tables we want to work with and how they're connected. For this, we can use `dm` or `as_dm()`. 
+
+`as_dm()` accepts lists of objects — as of this version, you must explicitly name the objects.
+
+```{r}
+flights_dm_no_keys <- as_dm(
+  list(
```
Comment (krlmlr) [inline vignettes/setting_up_dm.Rmd]: `tibble::lst()` would auto-name.

File: vignettes/setting_up_dm.Rmd [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+```{r}
+cdm_enum_fk_candidates(
+  dm = flights_dm_only_pks,
+  table = flights,
+  ref_table = airlines
+)
+```
+
+
+You define how your tables relate using `cdm_add_fk()`. First, define the tables you wish to connect to another using the function's options `table` and `ref_table`. Then indicate in `column` which column of `table` refers to `ref_table`'s primary key, which you've defined above. Use `check = FALSE` to coerce a key.
```
Comment (krlmlr) [inline vignettes/setting_up_dm.Rmd]: I don't understand "coerce". `check = FALSE` omits the consistency checks, ideally we want a notion of weak relationships (#4).

File: vignettes/setting_up_dm.Rmd [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+  table = planes
+  )
+```
+
+Now, add any primary keys you may have identified:
+
+```{r}
+flights_dm_only_pks <- flights_dm_no_keys %>%
+  cdm_add_pk(table = airlines, column = carrier) %>% 
+  cdm_add_pk(table = airports, column = faa) %>%
```
Comment (krlmlr) [inline vignettes/setting_up_dm.Rmd]: The explicit argument names are useful but also redundant. Maybe use in the first call and omit in the following calls? Same below for foreign keys.

File: vignettes/setting_up_dm.Rmd (hunk 3 above)
Comment (balthasars) [reply vignettes/setting_up_dm.Rmd]: ah I must have misunderstood what it does. Is changed now.

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**